README.md

```diff
@@ -50,154 +50,301 @@ Histolab has only one system-wide dependency: OpenSlide.
+subdirectory of the processedpath:
 
-print(f"Thumbnails saved at: {heart_slide.thumbnail_path}") 
+```
+prostate_slide.save_thumbnail()
+print(f"Thumbnails saved at: {prostate_slide.thumbnail_path}")
+prostate_slide.show()
 ```
 
-    Thumbnails saved at: processed/thumbnails/9c960533-2e58-4e54-97b2-8454dfb4b8c8.png
-    Thumbnails saved at: processed/thumbnails/JP2K-33003-2.png
+![](https://user-images.githubusercontent.com/4196091/92748324-5033e680-f385-11ea-812b-6a9a225ceca4.png)
 
+```
+ovarian_slide.save_thumbnail()
+print(f"Thumbnails saved at: {ovarian_slide.thumbnail_path}")
+ovarian_slide.show()
+```
 
+![](https://user-images.githubusercontent.com/4196091/92748248-3db9ad00-f385-11ea-846b-a5ce8cf3ca09.png)
 
-```python
-breast_slide.show() 
-heart_slide.show()
-```
+## Tile extraction
 
-![thumbnails](https://user-images.githubusercontent.com/31658006/84955475-a4695a80-b0f7-11ea-83d5-db7668801219.png)
+Once that the `Slide` objects are defined, we can proceed to extract the
+tiles. To speed up the extraction process, `histolab` automatically
+detects the tissue region with the largest connected area and crops the
+tiles within this field. The `tiler` module implements different
+strategies for the tiles extraction and provides an intuitive interface
+to easily retrieve a tile dataset suitable for our task. In particular,
+each extraction method is customizable with several common parameters:
 
-### Tiles extraction
+-   `tile_size`: the tile size;
+-   `level`: the extraction level (from 0 to the number of available
+    levels);
+-   `check_tissue`: if a minimum percentage of tissue is required to
+    save the tiles (default is 80%);
+-   `prefix`: a prefix to be added at the beginning of the tiles’
+    filename (default is the empty string);
+-   `suffix`: a suffix to be added to the end of the tiles’ filename
+    (default is .png).
 
-Now that your ```Slide``` object is defined, you can automatically extract the tiles. A ```RandomTiler``` object crops random tiles from the slide.
-You need to specify the size you want your tiles, the number of tiles to crop, and the level of magnification. If ```check_tissue``` is True, the exracted tiles are taken by default from the **biggest tissue region detected** in the slide, and the tiles are saved only if they have at least 80% of tissue inside.
+### Random Extraction
 
+The simplest approach we may adopt is to randomly crop a fixed number of
+tiles from our slides; in this case, we need the `RandomTiler`
+extractor:
 
-```python
+```
 from histolab.tiler import RandomTiler
+```
+
+Let us suppose that we want to randomly extract 6 squared tiles at level
+2 of size 512 from ourprostate slide, and that we want to save them only
+if they have at least 80% of tissue inside. We then initialize our
+`RandomTiler` extractor as follows:
+
+```
+PROSTATE_RANDOM_TILES_PATH = os.path.join(PROCESS_PATH_PROSTATE, 'random')# save tilesin the 'random' subdirectory
```

Comment:
```suggestion
PROSTATE_RANDOM_TILES_PATH = os.path.join(PROCESS_PATH_PROSTATE, 'random') # save tiles in the 'random' subdirectory
```

please break the line because it is too long